exe/Main.hs

```diff
@@ -406,25 +405,32 @@ loadSession dir = do
                     -- Keep the same name cache
                     modifyVar_ hscEnvs (return . Map.adjust (\(h, _) -> (h, [])) hieYaml )
                     consultCradle hieYaml cfp
-                  else return (opts, Map.keys old_di)
+                  else return (HM.keys v, (opts, Map.keys old_di))
               Nothing -> consultCradle hieYaml cfp
 
       -- The main function which gets options for a file. We only want one of these running
       -- at a time. Therefore the IORef contains the currently running cradle, if we try
       -- to get some more options then we wait for the currently running action to finish
       -- before attempting to do so.
-      let getOptions :: FilePath -> IO (IdeResult HscEnvEq, [FilePath])
+      let getOptions :: FilePath -> IO ([NormalizedFilePath],(IdeResult HscEnvEq, [FilePath]))
           getOptions file = do
               hieYaml <- cradleLoc file
               sessionOpts (hieYaml, file) `catch` \e ->
-                  return (([renderPackageSetupException compileTime file e], Nothing),[])
+                  return ([],(([renderPackageSetupException compileTime file e], Nothing),[]))
 
       returnWithVersion $ \file -> do
-        liftIO $ join $ mask_ $ modifyVar runningCradle $ \as -> do
+        (cs, opts) <- liftIO $ join $ mask_ $ modifyVar runningCradle $ \as -> do
           -- If the cradle is not finished, then wait for it to finish.
           void $ wait as
           as <- async $ getOptions file
-          return (as, wait as)
+          return (fmap snd as, wait as)
+        unless (null cs) $
+          void $ shakeEnqueueSession ideSession $ mkDelayedAction "InitialLoad" Info $ void $ do
+            cfps' <- liftIO $ filterM (IO.doesFileExist . fromNormalizedFilePath) cs
+            mmt <- uses GetModificationTime cfps'
+            let cs_exist = catMaybes (zipWith (<$) cfps' mmt)
+            uses GetModIface cs_exist
```

Comment:
This is done to get all errors in the project on startup, right? Please add a comment. 

What happens if the Shake session gets restarted? Is this action reenqueued?